File: src-tauri/src/dto/external/seventv.rs
use super::emote::EmoteEntry;
use serde::Serialize;

#[derive(Serialize)]
pub struct ChannelResult {
    pub emotes: Vec<EmoteEntry>,
    pub emote_set_id: Option<String>,
}

#[derive(Serialize, Clone)]
pub struct Delta {
    pub id: String,
    pub actor: Option<String>,
    pub added: Vec<EmoteEntry>,
    pub removed: Vec<String>,
    pub renamed: Vec<Rename>,
}

#[derive(Serialize, Clone)]
pub struct Rename {
    pub from: String,
    pub to: String,
}

File: src/lib/types/external/seventv.ts
import type { EmoteEntry } from "./emote.ts";

export type ChannelResult = {
  emotes: EmoteEntry[];
  emote_set_id: string | null;
};

export type Rename = {
  from: string;
  to: string;
};

export type Delta = {
  id: string;
  actor: string | null;
  added: EmoteEntry[];
  removed: string[];
  renamed: Rename[];
};
